lib/Doctrine/Query.php

```diff
@@ -1804,7 +1804,7 @@ public function getLimitSubquery()
                 // otherwise limit would be executed after ids are reordered by distinct / group by, and that order cannot be determined.
                 if ($driverName == 'mysql') {
                     // assign an incremented number to each row with id
-                    $subquery = 'SELECT @rownum:=@rownum + 1 as row_number,'
+                    $subquery = 'SELECT @rownum:=@rownum + 1 as `row_number`,'
```

Comment:
This should be quoted properly as follows:
```suggestion
                    $subquery = 'SELECT @rownum:=@rownum + 1 as ' . $this->_conn->quoteIdentifier('row_number') . ','
```
...then the failing test won't be failing anymore.